### ISSUE TITLE
bpmtk.jar au.edu.unimelb.services.ServiceProvider SMD wrong usage

### ISSUE THREAD

**Issue by iharsuvorau** [AdaptiveBProcess/Simod#14]
@Mcamargo85 

On macOS 11.4 with java:

```
java version "15.0.1" 2020-10-20
Java(TM) SE Runtime Environment (build 15.0.1+9-18)
Java HotSpot(TM) 64-Bit Server VM (build 15.0.1+9-18, mixed mode, sharing)
```

running `python simod_console.py` with a new java flag `-Xms1024M` instead of `-Xss8G` (see #13 for details) fails for another reason now:

```
 -- Mining Process Structure --
args ['java', '-Xmx2G', '-Xms1024M', '-cp', 'external_tools/splitminer3/bpmtk.jar:external_tools/splitminer3/lib/*', 'au.edu.unimelb.services.ServiceProvider', 'SMD', '0.5', '0.7', 'false', 'false', 'false', 'outputs/20210701_47B5EF4E_5884_472B_8C5F_CB8372D35250/PurchasingExample.xes', 'outputs/20210701_47B5EF4E_5884_472B_8C5F_CB8372D35250/PurchasingExample']
TESTCODE - SMD
ERROR: wrong usage.
RUN> java -cp bpmtk.jar;lib\* au.edu.unimelb.services.ServiceProvider SMD e n p 'logpath\log.[xes|xes.gz|mxml]' 'outputpath\outputname' 
PARAM: e = double in [0,1] : parallelism threshold (epsilon)
PARAM: n = double in [0,1] : percentile for frequency threshold (eta)
PARAM: p = [true|false] : replace non trivial OR joins?
EXAMPLE: java -cp bpmtk.jar;lib\* au.edu.unimelb.services.ServiceProvider SMD 0.1 0.4 .\logs\SEPSIS.xes.gz .\outputs\SEPSIS
java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
        at org.deckfour.xes.util.XsDateTimeConversion.parseXsDateTime(XsDateTimeConversion.java:96)
        at org.deckfour.xes.util.XsDateTimeConversionJava7.parseXsDateTime(XsDateTimeConversionJava7.java:129)
        at org.deckfour.xes.in.XesXmlParser$XesXmlHandler.startElement(XesXmlParser.java:292)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:518)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanStartElement(XMLDocumentFragmentScannerImpl.java:1386)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2725)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:541)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:324)
        at java.xml/javax.xml.parsers.SAXParser.parse(SAXParser.java:197)
        at org.deckfour.xes.in.XesXmlParser.parse(XesXmlParser.java:175)
        at com.raffaeleconforti.log.util.LogImporter.importFromInputStream(LogImporter.java:191)
        at com.raffaeleconforti.log.util.LogImporter.importFromFile(LogImporter.java:155)
        at au.edu.unimelb.services.ServiceProvider.SplitMinerService(ServiceProvider.java:522)
        at au.edu.unimelb.services.ServiceProvider.main(ServiceProvider.java:96)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 21 more
```

Is it the problem with `SMD e n p`? Even though `settings` dict contains valid number, 0.5 and 0.7.

**Comment by iharsuvorau** [comment]
@Mcamargo85 

Changing java version to:

```
openjdk version "1.8.0_275"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.275-b01, mixed mode)
```

helped to get further down, but still fails with:

```
parsing log, completed traces :: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 608/608 [00:01<00:00, 588.72it/s]
Rearranging log traces ...      [DONE]
Importance: DEBUG
Message: Start serializing log to XES.XML

Importance: DEBUG
Message: finished serializing log (2012.243896484375 msec.)

{'read_options': {'timeformat': '%Y-%m-%dT%H:%M:%S.%f', 'column_names': {'Case ID': 'caseid', 'Activity': 'task', 'lifecycle:transition': 'event_type', 'Resource': 'user', 'Start Timestamp': 'start_timestamp', 'Complete Timestamp': 'end_timestamp'}, 'one_timestamp': False, 'filter_d_attrib': True}, 'input': 'inputs', 'output': 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576', 'sm2_path': 'external_tools/splitminer2/sm2.jar', 'sm1_path': 'external_tools/splitminer/splitminer.jar', 'sm3_path': 'external_tools/splitminer3/bpmtk.jar', 'bimp_path': 'external_tools/bimp/qbp-simulator-engine.jar', 'align_path': 'external_tools/proconformance/ProConformance2.jar', 'aligninfo': 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/CaseTypeAlignmentResults.csv', 'aligntype': 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/AlignmentStatistics.csv', 'calender_path': 'external_tools/calenderimp/CalenderImp.jar', 'simulator': 'bimp', 'mining_alg': 'sm3', 'file': 'PurchasingExample.xes', 'repetitions': 1, 'simulation': True, 'sim_metric': 'tsd', 'add_metrics': ['day_hour_emd', 'log_mae', 'dl', 'mae'], 'concurrency': 0.0, 'epsilon': 0.5, 'eta': 0.7, 'alg_manag': 'replacement', 'gate_management': 'discovery', 'rp_similarity': 0.8, 'res_cal_met': 'discovered', 'arr_cal_met': 'discovered', 'res_dtype': '247', 'arr_dtype': '247', 'pdef_method': 'automatic', 'res_support': 0.05, 'res_confidence': 70, 'arr_support': 0.05, 'arr_confidence': 10}
 -- Mining Process Structure --
args ['java', '-Xmx2G', '-Xms1024M', '-cp', 'external_tools/splitminer3/bpmtk.jar:external_tools/splitminer3/lib/*', 'au.edu.unimelb.services.ServiceProvider', 'SMD', '0.5', '0.7', 'false', 'false', 'false', 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/PurchasingExample.xes', 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/PurchasingExample']
openjdk version "1.8.0_275"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.275-b01, mixed mode)
TESTCODE - SMD
LOGP - total events parsed: 15474
LOGP - start events parsed: 7737
LOGP - complete events parsed: 7737
LOGP - total distinct events: 23
LOGP - total distinct traces: 61
DEBUG - generating complex log
DFGP - settings (eta, epsilon, filter-type) > 0.5 : 0.7 : FWG
DEBUG - potential parallelisms: 0
DFGP - loops length TWO found: 3
DEBUG - removed parallelism edges: 0
eTIME - 0.376s
Loading of bpmn structure from file 100.0%...      [DONE]
evaluating train partition conformance:: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 449/449 [00:03<00:00, 117.51it/s]
complete traces: 449 , events: 7737
conformance percentage: 38.08%
evaluating conformance after replacement:: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 449/449 [00:03<00:00, 127.70it/s]
complete traces: 449 , events: 9551
conformance percentage: 100.0%
reading conformant training traces: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 449/449 [00:03<00:00, 130.88it/s]
-- Mining Simulation Parameters --
reading conformant training traces:: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 449/449 [00:03<00:00, 126.79it/s]
analysing resource pool:: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 745.86it/s]
mining calendars:: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:02<00:00,  1.10s/it]
extracting interarrivals:: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 84/84 [00:00<00:00, 4669.79it/s]
analysing gateways probabilities:: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 29/29 [00:00<00:00, 281562.07it/s]
mining tasks distributions:: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21/21 [00:03<00:00,  5.94it/s]
simulating:: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:03<00:00,  3.10s/it]
reading simulated logs:: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.03s/it]
[Errno 2] No such file or directory: 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/sim_data/PurchasingExample_1.csv'
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/Users/ihar/projects/api/simod/simod.py", line 251, in read_stats
    return read(*args)
  File "/Users/ihar/projects/api/simod/simod.py", line 240, in read
    rep_results = pd.read_csv(os.path.join(path, log_name),
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 610, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 462, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 819, in __init__
    self._engine = self._make_engine(self.engine)
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 1050, in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 1867, in __init__
    self._open_handles(src, kwds)
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/parsers.py", line 1362, in _open_handles
    self.handles = get_handle(
  File "/Users/ihar/projects/api/simod/venv/lib/python3.8/site-packages/pandas/io/common.py", line 647, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/sim_data/PurchasingExample_1.csv'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ihar/projects/api/simod/simod.py", line 65, in safety_check
    method(*args)
  File "/Users/ihar/projects/api/simod/simod.py", line 215, in simulate
    args = [(self.settings, self.process_stats, log) for log in p.get()]
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
FileNotFoundError: [Errno 2] No such file or directory: 'outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/sim_data/PurchasingExample_1.csv'
-- End of trial --
```

`outputs/20210701_C5B31B24_7FEE_4EEF_B8C6_CA989EBF8576/sim_data` is an empty folder for me

**Comment by Mcamargo85** [comment]
Issue solved with the change of Java version to 1.8
